clients/src/main/java/org/apache/kafka/clients/consumer/internals/PartitionAssignorAdapter.java

```diff
@@ -0,0 +1,129 @@
+
+    private static final Logger LOG = LoggerFactory.getLogger(PartitionAssignorAdapter.class);
+    private final PartitionAssignor oldAssignor;
+
+    PartitionAssignorAdapter(PartitionAssignor oldAssignor) {
+        this.oldAssignor = oldAssignor;
+    }
+
+    @Override
+    public ByteBuffer subscriptionUserData(Set<String> topics) {
+        return oldAssignor.subscription(topics).userData();
+    }
+
+    @Override
+    public GroupAssignment assign(Cluster metadata, GroupSubscription subscriptions) {
+        return oldToNewGroupAssignment(oldAssignor.assign(metadata, newToOldGroupSubscription(subscriptions)));
+    }
+
+    @Override
+    public void onAssignment(Assignment assignment, ConsumerGroupMetadata metadata) {
+        oldAssignor.onAssignment(oldToNewAssignment(assignment), metadata.generationId());
+    }
+
+    @Override
+    public String name() {
+        return oldAssignor.name();
+    }
+
+    private static PartitionAssignor.Assignment oldToNewAssignment(Assignment assignment) {
+        return new PartitionAssignor.Assignment(assignment.partitions(), assignment.userData());
+    }
+
+    private Map<String, PartitionAssignor.Subscription> newToOldGroupSubscription(GroupSubscription subscriptions) {
+        Map<String, PartitionAssignor.Subscription> oldSubscriptions = new HashMap<>();
+        for (Map.Entry<String, Subscription> entry : subscriptions.groupSubscription().entrySet()) {
+            String member = entry.getKey();
+            Subscription newSubscription = entry.getValue();
+            oldSubscriptions.put(member, new PartitionAssignor.Subscription(
+                newSubscription.topics(), newSubscription.userData()));
+        }
+        return oldSubscriptions;
+    }
+
+    private GroupAssignment oldToNewGroupAssignment(Map<String, PartitionAssignor.Assignment> assignments) {
+        Map<String, Assignment> newAssignments = new HashMap<>();
+        for (Map.Entry<String, PartitionAssignor.Assignment> entry : assignments.entrySet()) {
+            String member = entry.getKey();
+            PartitionAssignor.Assignment oldAssignment = entry.getValue();
+            newAssignments.put(member, new Assignment(oldAssignment.partitions(), oldAssignment.userData()));
+        }
+        return new GroupAssignment(newAssignments);
+    }
+
+    public static List<ConsumerPartitionAssignor> getAssignorInstances(List<String> classNames) {
+        List<ConsumerPartitionAssignor> assignorInstances = new ArrayList<>();
+        Class<ConsumerPartitionAssignor> newClass = ConsumerPartitionAssignor.class;
+        Class<PartitionAssignor> oldClass = PartitionAssignor.class;
+
+        if (classNames == null)
+            return assignorInstances;
+
+        for (Object klass : classNames) {
+            Object assignor;
+            if (klass instanceof String) {
+                try {
+                    assignor = Utils.newInstance((String) klass, newClass);
```

Comment:
I think we can use `Class.isAssignableFrom` to see what type it is rather than catching the exception. See `ChannelBuilders.createPrincipalBuilder` for a similar use case.

Reply:
Ack, cleaned up this method